### DISCUSSION TITLE
Why Redis?

### DISCUSSION THREAD

**Discussion by dross15** [caching-tools/next-shared-cache#518] [Q&A]
Hi! I was curious, why did you opt to build this out with Redis? I'm looking into implementing a local cache for a Next app but each static page (html) is roughly 100kB and we pre-generate around 5000 pages, so that'd be at least 0.5GB. In this case, s3 is much less expensive. I'm trying to wrap my head around the use case where Redis would be optimal? Maybe I'm missing something. Thanks!

**Comment by better-salmon** [comment]
Hi, @dross15! I'm grateful for your question. I chose Redis for its straightforwardness, speed, and APIs like full-text search and scanning that simplify on-demand revalidation. I opened [this issue](https://github.com/caching-tools/next-shared-cache/issues/123) a while ago, but my primary work and the fact that I'm the sole maintainer of this project have left me with insufficient time to implement it.

It would be beneficial to send HTML on a CDN with the correct cache-control header instead of storing it in Redis or raw S3 storage. We do this with static assets like JS and CSS, so why not HTML? CacheHandler will set and update HTML on the CDN where assets live. Partial prerendering is almost a thing now, and it helps to increase the lifetime of the HTML and store it as long as possible on a CDN.
